src/wines/wines.service.ts

```diff
@@ -67,4 +68,32 @@ export class WinesService {
     if (Wine === null) throw new NotFoundException();
     return this.wineRepository.remove(Wine);
   }
+
+  async update(id: string, updatedWine: CreateWineDto): Promise<Wine> {
+    const wine = await this.wineRepository.findOneOrFail({
+      where: { id },
+      relations: ['stores'],
+    });
+
+    if (!wine) throw new NotFoundException('Wine not found.');
```

Comment:
Instead of having these checks everywhere, take a look at this [approach](https://github.com/layla-oskar-schule/VinopinionsApi/blob/feature/friends/src/users/users.service.ts#L32). My idea here was to have a single location to handle an exception that would occur in many use cases and pass down the required relations as a parameter

Reply:
accidentaly pushed the changes to feature/ratings